### PR TITLE
Include Selective Check to Ping Maintainer about Legacy WWW and API Changes

### LINKED ISSUE #42031: Fail the PR if it modifies old "ui"
In case PR contains changes to the "old" uI ("airflow/www") and the PR does not have the "legacy UI" label set by the maintainer, the PR should fail. This is in order to limit changes that are implemented in the old UI to only bugfixes and necessary changes to enable "new UI" in Airflow 3.0.

This can be easily done via producing the right outputs in Breeze's selective checks.

* Breeze already has features to see if files in the incoming PR have belong to certain "file group" (and we have both "WWW" and "UI" groups defined
* Breeze already has access to labels that are assigned to PR

So it shoudl be as easy as simply exiting with error in selective checks and informing the user `Please ask maintainer to assign the "legacy UI" label to the PR in order to continue`

### PR DESCRIPTION
**Pull request by bugraoz93**
Closes: #42031
Include new check in selective_checks.py for checking PR labels to ping maintainers about updating legacy ui or legacy api code piece

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/utils/selective_checks.py [hunk 1]
```diff
@@ -157,6 +160,10 @@ def __hash__(self):
             r"^airflow/api_connexion/openapi/v1\.yaml",
             r"^clients/gen",
         ],
+        FileGroupForCi.LEGACY_API_FILES: [
+            r"^airflow/api/",
```
Comment (bugraoz93) [inline dev/breeze/src/airflow_breeze/utils/selective_checks.py]: On a second thought, maybe this should be deleted

File: dev/breeze/src/airflow_breeze/utils/selective_checks.py [hunk 2]
```diff
@@ -185,7 +191,7 @@ def __hash__(self):
             r"^airflow/ui/[^/]+\.json$",
             r"^airflow/ui/.*\.lock$",
         ],
-        FileGroupForCi.WWW_FILES: [
+        FileGroupForCi.LEGACY_WWW_FILES: [
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/selective_checks.py:194]: Cool. Legacy :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
I added labels.

Screenshot 2024-10-04 at 18 08 06

**Comment by potiuk** [comment]
BTW. @bugraoz93 -> one more small change could be to add a unit test for that -> we have a number of tests in selective checks - so maybe that case could also be unit-tested.

**Comment by bugraoz93** [comment]
Thanks @potiuk! I missed the unit test part. :sweat_smile: I created a task to follow up from there. I will include them soon :)

**Pull request merged**